### PR TITLE
Import WebDriver test harness files from wptrunner

### PR DESCRIPTION
**Pull request by jugglinmike**
Previously, WebDriver tests were executed against a WebDriver session
that was managed by the wptrunner tool. Beyond increasing coupling
between this project's tests and that particular test runner, this
factoring made it impossible to write tests against aspects of the
WebDriver protocol that explicitly concern session management.

Update the test logic to explicitly manage connections to the WebDriver
"remote end," (made possible by the runtime specification of host name
and port number), and import the relevant logic from the wptrunner
project.

This depends on https://github.com/w3c/wptrunner/pull/233.

### CODE REVIEW

File: webdriver/util/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (jugglinmike) [inline webdriver/util/cleanup.py]: @jgraham Is this license compatible with this project? If not, could we resolve this with a commit authored by you to remove it?

Comment (andreastt) [inline webdriver/util/cleanup.py]: I wrote this code. I hereby relinquish any rights I have on the code to the WPT project default license.

File: webdriver/conftest.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import os
+
+import pytest
+import webdriver
+
+import util.cleanup as cleanup
```
Comment (andreastt) [inline webdriver/conftest.py]: This seems redundant.

File: webdriver/conftest.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@pytest.fixture(scope="session")
+def _session(request):
+    host = os.environ.get("WD_HOST", default_host)
+    port = int(os.environ.get("WD_PORT", default_port))
+
+    session = webdriver.Session(host, port)
+
+    def destroy():
+        if session.session_id:
```
Comment (andreastt) [inline webdriver/conftest.py]: `is not None`

**Review by andreastt — CHANGES_REQUESTED** [review]

File: webdriver/util/cleanup.py (hunk 1 above)
Comment (jugglinmike) [inline webdriver/util/cleanup.py]: Ah, thanks!

File: webdriver/conftest.py (hunk 3 above)
Comment (jugglinmike) [inline webdriver/conftest.py]: Sorry, still getting a hang of that detail in Python.

File: webdriver/conftest.py (hunk 2 above)
Comment (jugglinmike) [inline webdriver/conftest.py]: Do you mean the general pattern (e.g. importing the module and defining the fixture here) or the specific aliasing (e.g. `as cleanup`)?

Comment (andreastt) [inline webdriver/conftest.py]: Sorry, I meant you could just do `from util import cleanup`.

**Comment by andreastt** [comment]
I also wonder if we should find a better name than `util`. I don’t want to create another kitchen sink.

**Comment by jugglinmike** [comment]
> I also wonder if we should find a better name than util. I don’t want to
> create another kitchen sink.

I'd suggest `fixtures/`, but I'm anticipating a few more utilities for future
tests. For instance, despite its small size, the `inline` function will be good
to share between many files. Another example is domain-specific assertions.

We could just dump them in the `webdriver/` directory, but I'm pretty averse to
mixing files containing tests with non-test files (even if Pytest is clever
enough to sort out the difference). Would you feel any better about something
like this:

    - webdriver/
      |`- tests/
      |   |`navigation.py
      |    `contexts.py
      |`- fixtures/
      |   `cleanup.py
      |`- inline.py
       `- assert.py

**Comment by jugglinmike** [comment]
@andreastt I know that my proposal above kind of makes the `webdriver/` directory itself a sort of "kitchen sink," but I'm open to suggestion for alternatives!

**Review by andreastt — APPROVED** [review]

**Comment by andreastt** [comment]
I think your proposal for document structure makes sense, although I would like the tests to be directly under /webdriver. They are the focus of attention and it should be clear to contributors where to find them.

I don’t however think we need to fix that immediately in this PR. It can come later.

**Comment by jugglinmike** [comment]
Thanks, Andreas!

**Pull request merged**